pypeit/flatfield.py

```diff
@@ -209,7 +213,107 @@ def slit_profile(self, slit):
+        self.slit_profiles
+
+        """
+
+        # Build the pixel flat (as needed)
+        if self.rawflatimg is None:
+            self.rawflatimg = self.build_pixflat()
+
+        # Prep tck (sets self.ntckx, self.ntcky)
+        self._prep_tck()
+
+        # Setup
+        self.mspixelflat = np.ones_like(self.rawflatimg)
+        self.msillumflat = np.ones_like(self.rawflatimg)
+        self.flat_model = np.zeros_like(self.rawflatimg)
+
+        # Loop on slits
+        for slit in range(self.nslits):
+            msgs.info("Computing flat field image for slit: {:d}".format(slit + 1))
+            thismask = (self.tslits_dict['slitpix'] == slit + 1)
+            if self.msbpm is not None:
+                inmask = thismask & ~self.msbpm
+            else:
+                inmask = thismask
+            # Fit flats for a single slit
+            self.mspixelflat[thismask], self.msillumflat[thismask], self.flat_model[thismask] = \
+                flat.fit_flat(self.rawflatimg, self.tilts, thismask,self.tslits_dict['lcen'][:, slit], self.tslits_dict['rcen'][:,slit],
+                              inmask = inmask, debug = debug)
+
+
+        if show:
+            # Global skysub is the first step in a new extraction so clear the channels here
+            self.show(slits=True, wcs_match = True)
+
+
+        # Return
+        return self.mspixelflat, self.msillumflat
+
+
+
+    def show(self, slits = True, wcs_match = True):
+
+        viewer, ch = ginga.show_image(self.mspixelflat, chname='pixeflat', cuts=(0.9, 1.1), wcs_match=wcs_match, clear=True)
+        viewer, ch = ginga.show_image(self.msillumflat, chname='illumflat', cuts=(0.9, 1.1), wcs_match=wcs_match)
+        viewer, ch = ginga.show_image(self.rawflatimg, chname='flat', wcs_match=wcs_match)
+        viewer, ch = ginga.show_image(self.flat_model, chname='flat_model', wcs_match=wcs_match)
+
+
+        if slits:
+            if self.tslits_dict is not None:
+                slit_ids = [trace_slits.get_slitid(self.rawflatimg.shape, self.tslits_dict['lcen'], self.tslits_dict['rcen'], ii)[0]
+                            for ii in range(self.tslits_dict['lcen'].shape[1])]
+                ginga.show_slits(viewer, ch, self.tslits_dict['lcen'], self.tslits_dict['rcen'], slit_ids)
+
+'''
+    def show_old(self, attr='norm', display='ginga'):
+        """
+        Show one of the internal images
+
+        Parameters
+        ----------
+        attr : str
+          mspixelflat -- Show the combined flat image, unnormalized
+          norm -- Show the combined normalized flat image
+        display : str, optional
+
+        Returns
+        -------
+
+        """
+        if attr == 'mspixelflat':
+            if self.mspixelflat is not None:
+                ginga.show_image(self.mspixelflat)
+        elif attr == 'norm':
+            if self.mspixelflatnrm is not None:
+                ginga.show_image(self.mspixelflatnrm)
+
+'''
+
+'''
+    def run_old(self, armed=False):
```

Comment:
Just remove this?

PS - The use and indentation of `'''` vs. `"""` is messy here is messy and another reason I prefer just using `#` on every line...

Reply:
I only use these quotations to comment out large code blocks. The reason I do that is because if I do what you suggest and add a # to every line, it is tedious to implement that with my editor (would have to do it line by line) and it can also lead to problems with the indentation. I've put this here to preserve the old code, in case we want to reference it.

Reply:
Okay, but honestly we really shouldn't even be thinking of use of `'''` as "commenting" code.  It actually turns the code into a docstring which sphinx may add to the documentation.  See here:

https://stackoverflow.com/questions/675442/comment-out-a-python-code-block

Reply:
Okay so should I be using """ instead, or is that the same. I agree that doc-strings are not the right approach. Unfortunately PyCharm does not have an option to comment out a sequence of lines, but I can try to look harder.